viewing_party/party.py

```diff
@@ -0,0 +1,212 @@
+    user_unique_list = []
+    for movie in friends_watched_movie_list:
+#bc friends_movie_list is compiled from multiple friends, there may be some duplicates in the list
+# need to add an additional check to make sure there are no duplicates in final unique_list
+        if movie not in user_data_movie_list:   #checking the list of title strings
+            for movie_dict in total_movie_dicts:  #checking all movies to find the corresponding dict to the title
+                if movie_dict['title'] == movie:    #comparing value of dict 'title' key (which is a string) to the title string from 
+#friends_watched_movie_list, which is a list of strings only.  Find a match which is how to access the whole dict, and not just the title string
+                    if movie_dict not in user_unique_list:  #make sure dict is not already in unique list
+                        user_unique_list.append(movie_dict)     #add dict to user_unique list
+    return user_unique_list #repair exploded brain and weep for the 12 hours it took to solve this
+
+# def get_friends_unique_watched(user_data): #1ST VERSION OF FUNCTION
+#     user_data_movie_list =[]
+#     for watched_dictionary in user_data['watched']:
+#         user_data_movie_list.append(watched_dictionary)
+# #this list will store all the movies (as dicts) from the 2 inner dicts from 'friends'
+#     friends_watched_movie_list =[]
+#     for watched_dictionary in user_data['friends']:
+#         for inner_dictionary in watched_dictionary['watched']:
+#             friends_watched_movie_list.append(inner_dictionary)
+# #this list will store just movies that friends have watched, but user has not
+#     user_unique_list = []
+#     for movie in friends_watched_movie_list:
+# #bc friends_movie_list is compiled from multiple friends, there may be some duplicates in the list
+# # need to add an additional check to make sure there are no duplicates in final unique_list
+#         if movie not in user_data_movie_list and movie not in user_unique_list:
+#             user_unique_list.append(movie)
+#     return user_unique_list
+
+
+
+    
+
+def get_available_recs(user_data):
+#call get_friends_unique-watched function to access user_unique_list
+#now have a list of dicts of the movies user wants to watch
+    user_wants_to_watch_list = get_friends_unique_watched(user_data)
+
+    recommended_movies = [] #recommended_movies is a list of dicts
+#this is a check to return an empty list if given an empty list
+    if len(user_wants_to_watch_list)== 0:
+        return recommended_movies
+#need to see if host value in user_wants_to_watch_list matches any of list of strings in 'subscriptions'
+#if there's a match, add dictionary to recommended_movies list
+    else:
+        for movie_dict in user_wants_to_watch_list:
+            if movie_dict['host'] in user_data['subscriptions']:
+                recommended_movies.append(movie_dict)
+    
+        return recommended_movies
+
+def get_new_rec_by_genre (user_data):
```

Comment:
you can also use `get_most_watched_genre(user_data) and get_unique_watched(user_data)` here